include/envoy/http/filter.h

```diff
@@ -27,7 +27,10 @@ enum class FilterHeadersStatus {
   // Do not iterate to any of the remaining filters in the chain. Returning
   // FilterDataStatus::Continue from decodeData()/encodeData() or calling
   // continueDecoding()/continueEncoding() MUST be called if continued filter iteration is desired.
-  StopIteration
+  StopIteration,
+  // Continue iteration to remaining filters, but ignore and subsequent data or trailers. This
+  // results in creating a header only request/response.
+  ContinueHeadersOnly
```

Comment:
I would prefer `ContinueAndEndStream` to HeadersOnly, matches gRPC term more (as it calls "Trailers-Only reponse"), and IMO clearer what it does. We may need `ContinueAndEndStream` for `FilterDataStatus` as well in future.

Reply:
Name change that @lizan proposed SGTM.